src/evalfunc.c

```diff
@@ -8700,8 +8718,32 @@ f_nr2char(typval_T *argvars, typval_T *rettv)
 
 	if (argvars[1].v_type != VAR_UNKNOWN)
 	    utf8 = (int)tv_get_number_chk(&argvars[1], NULL);
-	if (utf8)
-	    buf[(*utf_char2bytes)((int)tv_get_number(&argvars[0]), buf)] = NUL;
+	if (utf8 || enc_utf8)
+	{
+	    if (argvars[0].v_type == VAR_LIST)
+	    {
+		list_T		*l = argvars[0].vval.v_list;
+		listitem_T	*li;
+		int		i;
+		int		c;
+		int		len = 0;
+
+		if (l == NULL)
+		    return;
+		for (li = l->lv_first, i = 0;
+			  li != NULL && i < Screen_mco; li = li->li_next, ++i)
+		{
+		    c = tv_get_number(&li->li_tv);
+		    if (i > 0 && !utf_iscomposing(c))
+			break;
+		    len += utf_char2bytes(c, &buf[len]);
```

Comment:
Here, nr2char() stops on encountering a character that is not composing, but char2nr(str, 1, 1) does not stop in such a situation. IOW, I am not sure if it's good to have an asymetry: `char2nr("AB", 1, 1)` gives `[65, 66]`, but `nr2char([65, 66], 1)` gives only `"A"`.